### PR TITLE
DB-9846 Spark 3.0 Migrate minor API changes

### PR DESCRIPTION
**Pull request by ipraznik-splice**

### CODE REVIEW

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1097/

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1099/

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: hbase_sql/orc-spark-3.0/src/main/java/com/splicemachine/stream/output/ParquetWriterFactoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                        codec = CompressionCodecName.GZIP;
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Unknown compression: " + compression);
+                }
+                String extension = codec.getExtension() + ".parquet";
+                Path file = getDefaultWorkFile(taskAttemptContext, extension);
+                return getRecordWriter(conf, file, codec);
+            }
+        }.getRecordWriter(null);
```
Comment (martinrupp) [inline hbase_sql/orc-spark-3.0/src/main/java/com/splicemachine/stream/output/ParquetWriterFactoryImpl.java:80]: for other reviewers: the difference between this file and e.g. `hbase_sql/orc-spark-2.4/src/main/java/com/splicemachine/stream/output/ParquetWriterFactoryImpl.java` is

```
<         return encoder.createSerializer().apply(valueRow);
---
>         return encoder.toRow(valueRow);
```

File: hbase_sql/pom.xml [hunk 2]
```diff
@@ -297,7 +297,12 @@
             <version>${spark.version}</version>
             <scope>provided</scope>
         </dependency>
-
+        <dependency>
+            <groupId>org.apache.hive</groupId>
+            <artifactId>hive-classification</artifactId>
+            <version>${hive.version}</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (martinrupp) [inline hbase_sql/pom.xml:305]: sure we need this additional dependency?

File: splice_machine/src/main/java/com/splicemachine/derby/stream/control/ControlDataSet.java [hunk 3]
```diff
@@ -591,7 +592,7 @@ public String getAttribute(String name) {
                     ValueRow vr = (ValueRow) iterator.next();
                     context.recordWrite();
 
-                    rw.write(null, encoder.toRow(vr));
+                    rw.write(null, ParquetWriterService.getFactory().encodeToRow(tableSchema, vr, encoder));
```
Comment (martinrupp) [inline splice_machine/src/main/java/com/splicemachine/derby/stream/control/ControlDataSet.java:595]: for other reviewers: underlying change here is that Spark2.4 uses `encoder.toRow(valueRow);` , while Spark3.0 uses `return encoder.createSerializer().apply(valueRow);` . see `hbase_sql/orc-spark-3.0/src/main/java/com/splicemachine/stream/output/ParquetWriterFactoryImpl.java` and 2.4

File: standalone/spark3.0.0/src/test/resources/yarn-site.xml [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  ~ GNU Affero General Public License as published by the Free Software Foundation, either
+  ~ version 3, or (at your option) any later version.
+  ~ Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+  ~ without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  ~ See the GNU Affero General Public License for more details.
+  ~ You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+  ~ If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<configuration>
```
Comment (martinrupp) [inline standalone/spark3.0.0/src/test/resources/yarn-site.xml:15]: @OlegMazurov  might want to have a look at these configs.

**Comment by ipraznik-splice** [comment]
added a more detailed description of changes in Jira: https://splicemachine.atlassian.net/browse/DB-9846?focusedCommentId=68586

File: hbase_sql/pom.xml (hunk 2 above)
Comment (ipraznik-splice) [reply hbase_sql/pom.xml:305]: org.apache.hadoop.hive.serde2.SerDe is replaced by org.apache.hadoop.hive.serde2.AbstractSerDe. It needs this dependency for Public annotation to compile.

Comment (martinrupp) [reply hbase_sql/pom.xml:305]: ok

File: standalone/spark3.0.0/src/test/resources/yarn-site.xml (hunk 4 above)
Comment (OlegMazurov) [reply standalone/spark3.0.0/src/test/resources/yarn-site.xml:15]: Let's keep all yarn-site.xml in sync. If we decide to change some parameters for ITs/standalone that should be done in a separate JIRA.

File: hbase_sql/pom.xml (hunk 2 above)
Comment (OlegMazurov) [reply hbase_sql/pom.xml:305]: Add a comment to the dependency mentioning the required annotation to help future developers

File: hbase_sql/pom.xml [hunk 5]
```diff
@@ -297,7 +297,12 @@
             <version>${spark.version}</version>
             <scope>provided</scope>
         </dependency>
-
+        <dependency>
+            <groupId>org.apache.hive</groupId>
+            <artifactId>hive-classification</artifactId>
+            <version>${hive.version}</version>
```
[the remark this answers is not in the document]
Comment (ipraznik-splice) [reply hbase_sql/pom.xml:304]: I've added the comment to pom.xml

[the remark this answers is not in the document]
Comment (ipraznik-splice) [reply hbase_sql/pom.xml:304]: I've added the comment to pom.xml

[the remark this answers is not in the document]
Comment (ipraznik-splice) [reply hbase_sql/pom.xml:304]: I've added the comment to pom.xml

**Comment by ipraznik-splice** [comment]
> No need to make change to assembly to support a new spark version. It is only needed if you are to support a new CDH or HDP platform.

**Comment by ipraznik-splice** [comment]
> No need to make change to assembly to support a new spark version. It is only needed if you are to support a new CDH or HDP platform.

For test profile spark_3.0, I've set envClassifier to spark_3.0 as well. I needed this folder to create a parcel, but now the profile 'spark_3.0' removed. I've removed the corresponding folder.

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**